src/cmdhandler/command_handler.cpp

```diff
@@ -370,28 +440,218 @@ std::string command_handler::collector_system(const std::vector<std::string>& to
+                return "please provide a comma separated list of 9 doubles, "+std::to_string(values.size())+" given";
+            }
+
+            try{
+                if(variable == "camera_matrix"){
+                    current_state.camera.camera_matrix = values_by_comma_to_mat(values);
+                }else if(variable == "distortion_matrix"){
+                    current_state.camera.distortion_matrix = values_by_comma_to_mat(values);
+                }
+
+                return "'"+variable+"' variable set with values "+tokens[3];
+            }catch(const std::invalid_argument& err){
+                return "please provide a comma separated list of doubles";
+            }
+        }else if(variable == "marker_dictionary"){
+            if(tokens.size() != 4){
+                return "please provide an integer for variable '"+variable+"'\n    ex: set camera "+variable+" 6";
+            }
+
+            try{
+                current_state.camera.marker_dictionary = std::stoi(tokens[3]);
+            }catch(const std::invalid_argument& err){
+                return "please provide a valid integer value";
+            }
+
+            return "'"+variable+"' variable set with value "+tokens[3];
+        }else if(variable == "camera_options"){
+            if(tokens.size() != 5){
+                return "please provide a name and boolean value for '"+variable+"'\n    ex: set camera "+variable+" stream true";
+            }
+
+            std::string option_name = tokens[3];
+            bool option_value = (tokens[4] == "true");
+
+            //if the option already exists, update it, if not insert into camera_options map
+            auto index_found = current_state.camera.camera_options.find(option_name);
+            if(index_found != current_state.camera.camera_options.end()) {
+                index_found->second = option_value;
+            }else{
+                current_state.camera.camera_options.insert(std::pair<std::string, bool>(option_name, option_value));
+            }
+            return "'"+option_name+"' set to "+std::to_string(option_value);
+        }
+
+        return "variable '"+variable+"' does not exist";
+    }else if(tokens[0] == "get"){
+        if(tokens.size() < 3){
+            return "please provide a variable to get\n    ex: get camera url";
+        }
+
+        std::string variable = tokens[2];
+
+        if(variable == "url"){
+            return "url: "+current_state.camera.url;
+        }else if(variable == "camera_matrix" || variable == "distortion_matrix") {
+            std::stringstream response;
+            response << variable << ": ";
+
+            cv::Mat matrix_to_get;
+            if(variable == "camera_matrix"){
+                matrix_to_get = current_state.camera.camera_matrix;
+            }else if(variable == "distortion_matrix"){
+                matrix_to_get = current_state.camera.distortion_matrix;
+            }
+
+            for (int row = 0; row < matrix_to_get.rows; ++row) {
```

Comment:
We should probably have a utility function for printing out a matrix since we do it so often throughout the command handler